Source/Core/InputCommon/ControllerEmu.cpp

```diff
@@ -17,6 +17,14 @@ std::unique_lock<std::recursive_mutex> ControllerEmu::GetStateLock()
   return lock;
 }
 
+void ControllerEmu::RefreshDevices()
+{
+  ControllerInterface face;
+  UpdateReferences(face);
```

Comment:
I started off putting this logic in `InputConfigDiag`, so

```c++
// refresh devices
g_controller_interface.Reinitialize();
```

became

```c++
// refresh devices
ControllerInterface face;
controller->UpdateReferences(face);
g_controller_interface.Reinitialize();
controller->UpdateReferences(g_controller_interface);
```

This seemed like too much logic for DolphinWX, when the semantic action (refreshing devices while keeping a `ControlReference`) really belongs somewhere in InputCommon.